### PR TITLE
add cpack config to generate a debian package

### PR DESCRIPTION
**Pull request by akashters**
Some more setup is left to do like referencing dependencies (ex: libgtk2.0-dev).

But this setup can make a working installable deb package.
With enabled search in **launcher and icon**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -71,5 +75,19 @@ install(DIRECTORY ${ASSETS_DIR} DESTINATION ${CMAKE_INSTALL_DATADIR}/${PROJECT_N
...
 
 # CPack config
+set(CPACK_PACKAGE_NAME ${PROJECT_NAME})
+set(CPACK_PACKAGE_DESCRIPTION_SUMMARY "Chord naming application")
+set(CPACK_PACKAGE_VENDOR "Ters")
+set(CPACK_PACKAGE_VERSION_MAJOR ${PACKAGE_VERSION_MAJOR})
+set(CPACK_PACKAGE_VERSION_MINOR ${PACKAGE_VERSION_MINOR})
+set(CPACK_PACKAGE_VERSION_PATCH ${PACKAGE_VERSION_PATCH})
+set(CPACK_PACKAGE_ARCHITECTURE "amd64")
+set(CPACK_DEBIAN_PACKAGE_ARCHITECTURE "amd64")
```
Comment (shriramters) [inline CMakeLists.txt]: Do we need to set these values? I see cmake has some sensible defaults

**Comment by shriramters** [comment]
[CPack DMG](https://cmake.org/cmake/help/latest/cpack_gen/dmg.html#cpack_gen:CPack%20DragNDrop%20Generator)
[CPack DEB](https://cmake.org/cmake/help/latest/cpack_gen/deb.html#cpack_gen:CPack%20DEB%20Generator)
[CPack Variables](https://cmake.org/cmake/help/latest/module/CPack.html#variables-common-to-all-cpack-generators)
Check the defaults

File: CMakeLists.txt [hunk 2]
```diff
@@ -71,5 +75,19 @@ install(DIRECTORY ${ASSETS_DIR} DESTINATION ${CMAKE_INSTALL_DATADIR}/${PROJECT_N
 
 
 # CPack config
+set(CPACK_PACKAGE_NAME ${PROJECT_NAME})
+set(CPACK_PACKAGE_DESCRIPTION_SUMMARY "Chord naming application")
+set(CPACK_PACKAGE_VENDOR "Ters")
+set(CPACK_PACKAGE_VERSION_MAJOR ${PACKAGE_VERSION_MAJOR})
+set(CPACK_PACKAGE_VERSION_MINOR ${PACKAGE_VERSION_MINOR})
+set(CPACK_PACKAGE_VERSION_PATCH ${PACKAGE_VERSION_PATCH})
+set(CPACK_PACKAGE_ARCHITECTURE "amd64")
```
Comment (shriramters) [inline CMakeLists.txt]: couldn't find this variable in the documentation

File: CMakeLists.txt (hunk 1 above)
Comment (akashters) [reply CMakeLists.txt]: yeah, we just need to make sure build agent has dpkg installed to know the proper architecture

File: CMakeLists.txt (hunk 2 above)
Comment (akashters) [reply CMakeLists.txt]: got rid of it anyway

**Pull request merged**